presto-hive/src/main/java/io/prestosql/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -152,6 +156,7 @@ public static ParquetPageSource createParquetPageSource(
             TupleDomain<HiveColumnHandle> effectivePredicate,
             FileFormatDataSourceStats stats)
     {
+        checkState(columns.stream().allMatch(column -> column.getColumnType() == REGULAR), "column type must be regular");
```

Comment:
Considering this check may fail, would be good to include some debugging info:
```suggestion
        checkState(columns.stream().allMatch(column -> column.getColumnType() == REGULAR), "column type must be regular: %s", columns);
```